crypto/evp/evp_extra_test.cc

```diff
@@ -898,3 +1012,98 @@ TEST(EVPExtraTest, Ed25519Keygen) {
   ASSERT_TRUE(EVP_DigestVerify(ctx.get(), sig, len,
                                reinterpret_cast<const uint8_t *>("hello"), 5));
 }
+
+struct RsassaPssParamsMatchTestInput {
+  const uint8_t *der;
+  size_t der_len;
+  const EVP_MD *signature_md;
+  const EVP_MD *rsa_mgf1_md;
+  int match;
+} kRsassaPssParamsMatchTestInputs[] = {
+    {kExampleRSAPSSKeyPKCS8, sizeof(kExampleRSAPSSKeyPKCS8), EVP_sha1(),
+     EVP_sha1(), 0},
+    {kExampleRSAPSSKeyPKCS8, sizeof(kExampleRSAPSSKeyPKCS8), EVP_sha224(),
+     EVP_sha224(), 0},
+    {kExampleRSAPSSKeyPKCS8, sizeof(kExampleRSAPSSKeyPKCS8), EVP_sha256(),
+     EVP_sha256(), 1},
+    {kExampleRSAPSSKeyPKCS8, sizeof(kExampleRSAPSSKeyPKCS8), EVP_sha384(),
+     EVP_sha384(), 0},
+    {kExampleRSAPSSKeyPKCS8, sizeof(kExampleRSAPSSKeyPKCS8), EVP_sha512(),
+     EVP_sha512(), 0},
+    {kExampleRSAPSSKeyPKCS8, sizeof(kExampleRSAPSSKeyPKCS8), EVP_sha512_256(),
+     EVP_sha512_256(), 0},
+};
+
+class EVPRsaPssExtraTest
+    : public testing::TestWithParam<RsassaPssParamsMatchTestInput> {};
+
+// This test checks params match.
+// All parameters in the signature structure algorithm identifier MUST
```

Comment:
I'm not sure that is what you are testing here. I interpret 3.3 as talking about a key with an RSASSA-PSS algorithm identifier and a signature with an RSASSA-PSS algorithm identifier and then specifying how the associated parameters must match.